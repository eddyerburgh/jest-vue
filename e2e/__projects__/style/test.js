import { mount } from '@vue/test-utils'
import Stylus from './components/Stylus.vue'
import Scss from './components/Scss.vue'
import Sass from './components/Sass.vue'
import Less from './components/Less.vue'
import PostCss from './components/PostCss.vue'

test('processes Less', () => {
  const wrapper = mount(Less)
  expect(wrapper.is('div')).toBeTruthy()
})

test('processes PostCSS', () => {
  const wrapper = mount(PostCss)
  expect(wrapper.is('section')).toBeTruthy()
  expect(wrapper.vm.$style.red).toEqual('red')
  expect(wrapper.html()).toMatchSnapshot()
})

test('processes Sass', () => {
  const wrapper = mount(Sass)
  expect(wrapper.vm.$style.a).toEqual('a')
  expect(wrapper.vm.$style.b).toEqual('b')
  expect(wrapper.vm.$style.c).toEqual('c')
  expect(wrapper.vm.$style.light).toBeUndefined()
})

test('processes SCSS with resources', () => {
  const wrapper = mount(Scss)
  expect(wrapper.vm.$style.a).toEqual('a')
  expect(wrapper.vm.$style.b).toEqual('b')
  expect(wrapper.vm.$style.c).toEqual('c')
})

test('process Stylus', () => {
  const wrapper = mount(Stylus)
  expect(wrapper.vm).toBeTruthy()
})
