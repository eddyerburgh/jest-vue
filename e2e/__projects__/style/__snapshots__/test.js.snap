// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processes PostCSS 1`] = `"<section><div></div> <div class=\\"red\\"></div></section>"`;
